### PR TITLE
added wait to hitRateLimit and updated readme

### PR DESCRIPTION
**Pull request by deweydbb**

### CODE REVIEW

File: src/limiter/limiter.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func (l *Limiter) RequestSuccessful(requestWeight int) error {
 //the RateLimitConfig in the Limiter struct to prevent more 429s in the future.
 //The wait parameter is optional. It is the amount of time in milliseconds after HitRateLimit is
 //called that no requests will be allowed.
-func (l *Limiter) HitRateLimit(requestWeight int) error {
+func (l *Limiter) HitRateLimit(requestWeight int, wait ...int64) error {
```
Comment (dkoston) [inline src/limiter/limiter.go]: what's the thought around `...int64` here? Not seeing the use case so hard to follow

File: src/limiter/limiter.go [hunk 2]
```diff
@@ -165,7 +165,11 @@ func (l *Limiter) HitRateLimit(requestWeight int) error {
 			return err
 		}
 
-		if err = l.adjustConfig(requestWeight, c); err != nil {
+		if len(wait) == 0 {
+			wait = append(wait, 0)
```
Comment (dkoston) [inline src/limiter/limiter.go]: this seems "hacky". I don't understand the use case for `...int64` but it seems like since you're calling `wait[0]` below this really shouldn't be variadic. The logic seems to be like:

- If nothing is passed for wait, don't wait
- If something is passed for wait, do wait.

It seems like the amount to wait should be handled once when creating the config, rather than each time this is called as the caller would have to have access to the config to know how long to wait.

**Review by dkoston — CHANGES_REQUESTED** [review]

File: src/limiter/limiter.go (hunk 1 above)
Comment (deweydbb) [reply src/limiter/limiter.go]: the ...int64 makes it optional. The use case is that different APIs lock you out for different period of times after you hit the rate limit. For example I believe that kraken suspends your requests for fifteen minutes after hitting their rate limit.

Comment (dkoston) [reply src/limiter/limiter.go]: Ok. Mentioned this below but I feel like that belongs in the config. It’s not going to change after the config is set so it seems odd to pass it into this function versus referencing the config when the rate limit is hit.

File: src/limiter/requestsstatus.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ func (r *RequestsStatus) canMakeRequestLogic(requestWeight int, config RateLimit
 	now := getUnixTimeMilliseconds()
 
 	timeSinceLastError := now - r.lastErrorTime
-	if timeSinceLastError < config.timePeriod*1000 {
+	if timeSinceLastError < config.waitAfterHitLimit*1000 {
```
Comment (dkoston) [inline src/limiter/requestsstatus.go]: nice, this change makes this logic here more accurate/useful

**Comment by dkoston** [comment]
Thanks for the changes. I believe that part of the config was the missing logic in this library. These changes keep the API clean and easy to use. LGTM
